src/fft.jl

```diff
@@ -0,0 +1,576 @@
+using AbstractFFTs
+using LinearAlgebra
+using Dagger: DArray, @spawn, InOut, In
+using MPI
+
+struct FFT end
+struct RFFT end
+struct IRFFT end
+struct IFFT end
+struct FFT! end
+struct RFFT! end
+struct IRFFT! end
+struct IFFT! end
+
+export FFT, RFFT, IRFFT, IFFT, FFT!, RFFT!, IRFFT!, IFFT!, fft, ifft
+abstract type Decomposition end
+struct Pencil <: Decomposition end
+struct Slab <: Decomposition end
+
+function plan_transform(transform, A, dims; kwargs...)  
+        if transform isa RFFT
+            return plan_rfft(A, dims; kwargs...)
+        elseif transform isa FFT
+            return plan_fft(A, dims; kwargs...)
+        elseif transform isa IRFFT
+            return plan_irfft(A, dims; kwargs...)
+        elseif transform isa IFFT
+            return plan_ifft(A, dims; kwargs...)
+        elseif transform isa FFT!
```

Comment:
```suggestion
        if transform isa FFT!
```

We can't efficiently handle returning values from Dagger tasks when using Datadeps (`spawn_datadeps`) - instead we should pre-allocate memory and pass it to tasks. Thus, `plan_transform` should only operate in-place.